pilot/pkg/serviceregistry/aggregate/controller.go

```diff
@@ -144,19 +144,6 @@ func (c *Controller) Services() ([]*model.Service, error) {
 					sp.ClusterVIPs = make(map[string]string)
 				}
 				sp.ClusterVIPs[r.Cluster()] = s.Address
-
-				if s.Attributes.ClusterExternalAddresses != nil && len(s.Attributes.ClusterExternalAddresses[r.Cluster()]) > 0 {
-					if sp.Attributes.ClusterExternalAddresses == nil {
-						sp.Attributes.ClusterExternalAddresses = make(map[string][]string)
-					}
-					sp.Attributes.ClusterExternalAddresses[r.Cluster()] = s.Attributes.ClusterExternalAddresses[r.Cluster()]
-				}
-				if s.Attributes.ClusterExternalPorts != nil && len(s.Attributes.ClusterExternalPorts[r.Cluster()]) > 0 {
-					if sp.Attributes.ClusterExternalPorts == nil {
-						sp.Attributes.ClusterExternalPorts = make(map[string]map[uint32]uint32)
-					}
-					sp.Attributes.ClusterExternalPorts[r.Cluster()] = s.Attributes.ClusterExternalPorts[r.Cluster()]
-				}
```

Comment:
Why remove this? the data for a service returned by both services() and getService() should be consistent. Besides this is not heavy computation right? It’s just a cheap map merging.

Reply:
This is a legacy bug, we only GetService when calc gateway address

Reply:
The external port and address only used for gateway address cal

Reply:
That is true.. but for future use, any caller expecting to see the cluster external address in the service object will be confused as to why it is not appearing. This is not a costly merge right? So does it matter if we have this logic here or not?

Reply:
I meant to merge Services() and GetService() once i am free

Reply:
But if you insist on, I can restore it

Reply:
never mind about this code for now.